sdl2-sys/build.rs

```diff
@@ -273,20 +273,15 @@ fn generate_bindings<S: AsRef<str> + ::std::fmt::Debug>(target: &str, host: &str
     };
 
     // SDL2 hasn't a default configuration for Linux
-    if target_os == "linux" {
+    if target_os == "linux-gnu" {
         bindings = bindings.clang_arg("-DSDL_VIDEO_DRIVER_X11");
         bindings = bindings.clang_arg("-DSDL_VIDEO_DRIVER_WAYLAND");
     }
 
     let bindings = bindings
         .header("wrapper.h")
-        .hide_type("_")
-        .hide_type("FP_NAN")
-        .hide_type("FP_INFINITE")
-        .hide_type("FP_ZERO")
-        .hide_type("FP_SUBNORMAL")
-        .hide_type("FP_NORMAL") // Until https://github.com/rust-lang-nursery/rust-bindgen/issues/687 gets fixed
-        .hide_type("max_align_t") // Until https://github.com/rust-lang-nursery/rust-bindgen/issues/550 gets fixed
+        .rustified_enum(".*")
```

Comment:
According to https://docs.rs/bindgen/0.35.0/bindgen/struct.Builder.html#method.rustified_enum we should be warry when using this.

> You should not be using Rust enums unless you have complete control of the C/C++ code that you're binding to.

Since the SDL2 headers and source code may change in the future (especially enums), can you explain why we can safely use "rustified_enum" here? This is more a question than an actual remark, I'm trying to understand what this does exactly and what the impact may be.

Reply:
What happened when I updated bindgen was that some types previously generated as enums like SDL_EventType was now being generated as "unwrapped" constants, so I started googling why that was and found these issues:
https://github.com/rust-lang-nursery/rust-bindgen/issues/758
https://github.com/rust-lang-nursery/rust-bindgen/issues/667
https://github.com/rust-lang-nursery/rust-bindgen/issues/225

From what I understand not all C enums can be generated as rust enums, and it might depend on their usage whether they can or not. The reason why I added the rustified_enum method here was to not break the old API regarding those types, because then I had to update tons of places were the enum types in SDL was used to now use constants instead.
I do not know whether it is safe to continue generating the types as enum or not as I did not quite understand all of the issue. It seemed to be related to some nonzero optimization, and enums without a zero value, however I'm not sure why bindgen can't just detect whether the enum has a zero value or not, but perhaps it's a complicated issue. For the users of bindgen at least the effect is that previously generated enums are not generated as enums anylonger, it seems like one have to specify which one to generate as enums.

As a side note bindgen has added a feature which might be what one might want in some of the modules like mixer, there is a way to specify that an enum should be generated as bitflags, however from the looks of it the majority in the core SDL seem to not be bitflags.